applications/GeoMechanicsApplication/custom_elements/geo_truss_element.cpp

```diff
@@ -0,0 +1,305 @@
+                                           temp_process_information);
+        Vector temp_strain = ZeroVector(1);
+        temp_strain[0] = this->CalculateGreenLagrangeStrain();
+        Values.SetStrainVector(temp_strain);
+
+        array_1d<double, 3>  temp_internal_stresses = ZeroVector(3);
+        mpConstitutiveLaw->CalculateValue(Values, FORCE, temp_internal_stresses);
+
+        for (unsigned int i=0; i<TDim; ++i)
+            temp_internal_stresses[i] += mInternalStressesFinalizedPrevious[i];
+
+        rOutput[0] = temp_internal_stresses;
+    }
+    if (rVariable == CAUCHY_STRESS_VECTOR) {
+
+        ProcessInfo temp_process_information;
+
+        ConstitutiveLaw::Parameters Values(this->GetGeometry(),
+                                           this->GetProperties(),
+                                           temp_process_information);
+        Vector temp_strain = ZeroVector(1);
+        temp_strain[0] = this->CalculateGreenLagrangeStrain();
+        Values.SetStrainVector(temp_strain);
+        array_1d<double, 3>  temp_internal_stresses = ZeroVector(3);
+        mpConstitutiveLaw->CalculateValue(Values, FORCE, temp_internal_stresses);
+
+        for (unsigned int i=0; i<TDim; ++i)
+            temp_internal_stresses[i] += mInternalStressesFinalizedPrevious[i];
+
+        const double l = GeoStructuralMechanicsElementUtilities::CalculateCurrentLength3D2N(*this);
+        const double L0 = GeoStructuralMechanicsElementUtilities::CalculateReferenceLength3D2N(*this);
+
+        rOutput[0] = temp_internal_stresses*l/L0;
+    }
+
+    KRATOS_CATCH("")
+}
+
+//----------------------------------------------------------------------------------------
+template< unsigned int TDim, unsigned int TNumNodes >
+void GeoTrussElement<TDim,TNumNodes>::
+    CalculateOnIntegrationPoints(const Variable<array_1d<double, 3>>& rVariable,
+                                 std::vector<array_1d<double, 3>>& rOutput,
+                                 const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    const GeometryType::IntegrationPointsArrayType&
+        integration_points = this->GetGeometry().IntegrationPoints();
+    if (rOutput.size() != integration_points.size()) {
+        rOutput.resize(integration_points.size());
+    }
+
+    if (rVariable == FORCE) {
+        BoundedVector<double, TDim> truss_forces = ZeroVector(TDim);
+        truss_forces[2] = 0.00;
+        truss_forces[1] = 0.00;
+        const double A = this->GetProperties()[CROSS_AREA];
+
+        double prestress = 0.00;
+        if (this->GetProperties().Has(TRUSS_PRESTRESS_PK2)) {
+            prestress = this->GetProperties()[TRUSS_PRESTRESS_PK2];
+        }
+
+        const double L0 = GeoStructuralMechanicsElementUtilities::CalculateReferenceLength3D2N(*this);
+        const double l = GeoStructuralMechanicsElementUtilities::CalculateCurrentLength3D2N(*this);
+
+        ProcessInfo temp_process_information;
```

Comment:
this is very bad, please don't use temp instances of ProcessInfo any more
it is not necessary any more since now the processinfo is always passed as const

Reply:
Processinfo is passed.